modAionImpl/src/org/aion/zero/impl/sync/TaskGetHeaders.java

```diff
@@ -70,22 +75,31 @@
         this.log = log;
     }
 
+    /** Checks that the peer's total difficulty is higher than the local chain. */
+    private boolean isHigherTotalDifficulty(INode n) {
+        return n.getTotalDifficulty() != null && n.getTotalDifficulty().compareTo(this.selfTd) >= 0;
```

Comment:
is the ">=" an expecting behavior? cause the method name is higher than TD

Reply:
Technically, since there is a time delay between receiving the peer status and performing this check, a peer with a total difficulty equal to the local one has likely already advanced to a higher TD.
Still, I will rename the method to make it less ambiguous.

Reply:
updated in 22a2d69